scripts/nb-tester/test-notebook.py

```diff
@@ -0,0 +1,93 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may obtain a
+# copy of this license in the LICENSE file in the root directory of this source
+# tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this copyright
+# notice, and modified files need to carry a notice indicating that they have
+# been altered from the originals.
+
+import sys
+from pathlib import Path
+import nbconvert
+import nbformat
+
+WRITE_FLAG = "--write"
+NOTEBOOKS_GLOB = "docs/**/*.ipynb"
+NOTEBOOKS_EXCLUDE = [
+    "docs/api/**",
+    "**/.ipynb_checkpoints/**",
+    # Following notebooks have code errors
+    "docs/build/circuit-construction.ipynb",
+    "docs/build/pulse.ipynb",
+    "docs/transpile/custom-transpiler-pass.ipynb",
+    "docs/transpile/transpiler-stages.ipynb",
+    # Following notebook has some non-python dependency I can't figure out
+    "docs/build/circuit-visualization.ipynb",
+    # Following notebooks make requests so can't be tested yet
```

Comment:
Idk how we're going to test this. But I really like that you're not letting it get in the way. Testing 80% of notebooks is better than 0%.